packages/lodestar/src/sync/reqResp/reqResp.ts

```diff
@@ -97,7 +98,25 @@ export class SyncReqResp implements ISyncReqResp {
     } catch (e) {
       this.network.reqResp.sendResponse(id, e, null);
     }
-    // TODO handle incorrect forkVersion or disjoint finalizedCheckpoint
+    if (await this.shouldDisconnectOnHello(peerInfo, request)) {
+      this.network.reqResp.goodbye(peerInfo, 0n);
+    }
+  }
+
+  public async shouldDisconnectOnHello(peerInfo: PeerInfo, request: Hello): Promise<boolean> {
+    const headBlock = await this.db.block.getChainHead();
+    if (!headBlock) {
+      return true;
```

Comment:
We should always have chain head at this point. Might be better to throw exception as we shouldn't proceed after this,.

Reply:
Ah I had to do that earlier because of my unit test, now we don't have that anymore.
Let's remove it to make it consistent to other places.